spec-tests/src/test/java/org/candlepin/spec/OwnerProductResourceSpecTest.java

```diff
@@ -0,0 +1,799 @@
+        assertBadRequest(() -> ownerProductApi.deleteProductByOwner(owner.getKey(), providedProduct.getId()));
+    }
+
+    @Test
+    @OnlyInHosted
+    @DisplayName("should return forbidden request on attempt to delete provided product attached to sub")
+    public void shouldReturnForbiddenRequestOnAttemptToDeleteProvidedProductAttachedToSub()
+        throws Exception {
+        OwnerDTO owner = ownerApi.createOwner(Owners.random());
+        ProductDTO product = createHostedProductWithProvidedAndDerivedProduct();
+        ProductDTO providedProduct = Iterables.getOnlyElement(product.getProvidedProducts());
+        createUpstreamSubscription(owner, product);
+        assertForbidden(() -> ownerProductApi.deleteProductByOwner(owner.getKey(), providedProduct.getId()));
+    }
+
+    @Test
+    @OnlyInStandalone
+    @DisplayName("should return bad request on attempt to delete derived product attached to sub")
+    public void shouldReturnBadRequestOnAttemptToDeleteDerivedProductAttachedToSub() throws Exception {
+        OwnerDTO owner = ownerApi.createOwner(Owners.random());
+        ProductDTO product = createProductWithProvidedAndDerivedProduct(owner.getKey());
+        ProductDTO derivedProduct = product.getDerivedProduct();
+        assertBadRequest(() -> ownerProductApi.deleteProductByOwner(owner.getKey(), derivedProduct.getId()));
+    }
+
+    @Test
+    @OnlyInHosted
+    @DisplayName("should return forbidden request on attempt to delete derived product attached to sub")
+    public void shouldReturnForbiddenRequestOnAttemptToDeleteDerivedProductAttachedToSub()
+        throws Exception {
+        OwnerDTO owner = ownerApi.createOwner(Owners.random());
+        ProductDTO product = createHostedProductWithProvidedAndDerivedProduct();
+        ProductDTO derivedProduct = product.getDerivedProduct();
+        createUpstreamSubscription(owner, product);
+        assertForbidden(() -> ownerProductApi.deleteProductByOwner(owner.getKey(), derivedProduct.getId()));
+    }
+
+    @Test
+    @OnlyInStandalone
+    @DisplayName("should return bad request on attempt to delete derived provided product attached to sub")
+    public void shouldReturnbadRequestOnAttemptToDeleteDerivedProvidedProductAttachedToSub()
+        throws Exception {
+        OwnerDTO owner = ownerApi.createOwner(Owners.random());
+        ProductDTO product = createProductWithProvidedAndDerivedProduct(owner.getKey());
+        ProductDTO derivedProvidedProduct =
+            Iterables.getOnlyElement(product.getDerivedProduct().getProvidedProducts());
+        assertBadRequest(() -> ownerProductApi
+            .deleteProductByOwner(owner.getKey(), derivedProvidedProduct.getId()));
+    }
+
+    @Test
+    @OnlyInHosted
+    @DisplayName("should return forbidden request when deleting derived provided product attached to sub")
+    public void shouldReturnForbiddenRequestWhenDeletingDerivedProvidedProductAttachedToSub()
```

Comment:
This test can be generalized with custom products and pools.

Reply:
Updated